### PR TITLE
Identify and eliminate TaskModel Thread 

### LINKED ISSUE #1202: Identify and eliminate TaskModel Thread
TaskModel Thread has not unique name to identified them. Also, there are leakage not closed properly. For example in estJobFailureTaskNotStarted. Identify the cases and eliminate this pattern of leakage.

### PR DESCRIPTION
**Pull request by kaisun2000**
### Issues

- [x] My PR addresses the following Helix issues and references them in the PR description:

Fix #1202 

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

HelixTaskExcutor thread pool has not unqiue name. Assign unique
name. Also, for TaskModelFactory related test, there are leaks
in a test. Eliminate the leak.

### Tests

- [ ] The following tests are written for this issue:

(List the names of added unit/integration tests)

- [x] The following is the result of the "mvn test" command on the appropriate module:

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR] org.apache.helix.controller.strategy.crushMapping.TestCardDealingAdjustmentAlgorithmV2.testComputeMappingForDifferentReplicas(org.apache.helix.controller.strategy.crushMapping.TestCardDealingAdjustmentAlgorithmV2)
[ERROR]   Run 1: TestCardDealingAdjustmentAlgorithmV2.testComputeMappingForDifferentReplicas:273 Total movements: 4 != expected 8, replica: 1
[ERROR]   Run 2: TestCardDealingAdjustmentAlgorithmV2.testComputeMappingForDifferentReplicas:273 Total movements: 4 != expected 8, replica: 2
[ERROR]   Run 3: TestCardDealingAdjustmentAlgorithmV2.testComputeMappingForDifferentReplicas:273 Total movements: 12 != expected 21, replica: 3
[INFO]   Run 4: PASS
[INFO]   Run 5: PASS
[INFO] 
[ERROR]   TestEnableCompression.testEnableCompressionResource:117 expected:<true> but was:<false>
[ERROR]   TestTaskRebalancerStopResume.stopDeleteJobAndResumeNamedQueue:199 » Helix Fail...
[INFO] 
[ERROR] Tests run: 1151, Failures: 3, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:20 h
[INFO] Finished at: 2020-08-05T19:00:41-07:00
[INFO] ------------------------------------------------------------------------

Individual run would pass

### Commits

- [ ] My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation (Optional)

- [ ] In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Code Quality

- [ ] My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Comment by kaisun2000** [comment]
This diff is approved. Please help to merge into trunk.

>HelixTaskExcutor thread pool has not unqiue name. Assign unique
name. Also, for TaskModelFactory related test, there are leaks
in a test. Eliminate the leak.

**Comment by jiajunwang** [comment]
I can help to merge once the test results are post : ) @kaisun2000

**Comment by kaisun2000** [comment]
> I can help to merge once the test results are post : ) @kaisun2000

The test result posted. Mostly passed. Failed one can pass by running individually.

File: helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java [hunk 1]
```diff
@@ -209,7 +211,7 @@ public void registerMessageHandlerFactory(String type, MessageHandlerFactory fac
     if (prevItem == null) {
       ExecutorService newPool = Executors.newFixedThreadPool(threadpoolSize, new ThreadFactory() {
         @Override public Thread newThread(Runnable r) {
-          return new Thread(r, "HelixTaskExecutor-message_handle_thread");
+          return new Thread(r, "HelixTaskExecutor-message_handle_thread_" + thread_uid.getAndIncrement());
```
Comment (mgao0) [inline helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java:214]: Not related to your change. I'm wondering if we should make this daemon thread? @jiajunwang

**Comment by kaisun2000** [comment]
The pull request is approved, please help to merge into trunk.

>HelixTaskExcutor thread pool has not unqiue name. Assign unique
name. Also, for TaskModelFactory related test, there are leaks
in a test. Eliminate the leak.

File: helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java (hunk 1 above)
Comment (kaisun2000) [reply helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java:214]: Daemon thread is just to make sure that JVM can shutdown automatically when main thread ( and all other non daemon thread) exit. It is similar to you Unix machine restart, daemon process won't block it from restarting, but your forground process, say intellij can block the restart of Unix machine.

**Pull request merged**